packages/@aws-cdk/aws-lambda-nodejs/lib/package-manager.ts

```diff
@@ -39,7 +39,7 @@ export class PackageManager {
       case LockFile.PNPM:
         return new PackageManager({
           lockFile: LockFile.PNPM,
-          installCommand: logLevel && logLevel !== LogLevel.INFO ? ['pnpm', 'install', '--reporter', 'silent'] : ['pnpm', 'install'],
+          installCommand: logLevel && logLevel !== LogLevel.INFO ? ['pnpm', 'install', '--reporter', 'silent', '--shamefully-hoist'] : ['pnpm', 'install', '--shamefully-hoist'],
```

Comment:
Pretend I don't know shit about pnpm here (because I don't). What is `--shamefully-hoist` doing here?

Reply:
I've changed from `--shamefully-hoist` to use a `.npmrc` file, I think this way is better

I have explained what it does in the PR description

Reply:
I'm now using `--config.node_linker=hoisted`, as explained in the PR description